### ISSUE TITLE
Only change the system theme when device is locked

### ISSUE THREAD

**Issue by vixalien** [AutoDarkMode/Windows-Auto-Night-Mode#371]
**Describe the enhancement or feature you'd like**
On Samsungs, the device will only change the system theme after it has been locked (i.e. the screen is off.) This is because switching themes while working is somehow jarring to see and so Samsung naturally wait for the screen to be off to apply the new theme.

**Describe alternatives you've considered**
It would be nice to have that as an option in system settings like: "Wait for screen to be off before applying new theme"

**Additional context**
However there is a problem: Window devices rarely go to sleep while Phones' screens usually timeout in 30 seconds. I think it would be better to wait for inactivity of the user (i.e. not exactly when screen is off.) like if the user hasn't been active for 1 min or if there is a screensaver on (or the dark color that appears when you're not using your PC).

**Comment by Spiritreader** [comment]
Sounds doable.
The Windows API has a function where you can determine whether user inputs were made in the last `x` seconds.

**Comment by carlosvsilva** [comment]
This is exactly what I was about to suggest too, your software should have a toggable option of "Switch Dark Mode only after user is away from the computer for 5-15 minutes".
Because otherwise having to go through the Dark Mode switch in real-time, it takes a long time, and sometimes users are nearly hitting a professional deadline, and can't for the life of them be interrupted by the Windows Dark Mode switch.

Please give serious consideration to this improvement. Great software by the way!

Best regards,
Carlos

**Comment by Spiritreader** [comment]
> This is exactly what I was about to suggest too, your software should have a toggable option of "Switch Dark Mode only after user is away from the computer for 5-15 minutes". Because otherwise having to go through the Dark Mode switch in real-time, it takes a long time, and sometimes users are nearly hitting a professional deadline, and can't for the life of them be interrupted by the Windows Dark Mode switch.
> 
> Please give serious consideration to this improvement. Great software by the way!
> 
> Best regards, Carlos

Serious consideration is given, it's on the short list of new features to implement!

**Comment by Spiritreader** [comment]
available in latest beta!
